conn.go

```diff
@@ -482,6 +490,7 @@ func (c *Conn) loop(ctx context.Context) {
 		}
 
 		c.setState(StateDisconnected)
+		disconnectTime = time.Now()
```

Comment:
i think this should place after line 489;

when zookeeper cluster re-elect, reconnect always success, but auth failed:(EOF).  line 493 will rfresh disconnect every loop;  line 442 always be False